pkg/utils/utils.go

```diff
@@ -113,3 +113,18 @@ func HasUserspaceDriver(pciAddr string) (bool, error) {
 	}
 	return false, nil
 }
+
+func GetBridgeInterfaces(bridgename string) (brifs []string, err error) {
+	list := filepath.Join(NetDirectory, bridgename, "brif")
```

Comment:
As alternative we can use netlink based implementation. We can pass Bridge netlink struct to the function and then use netlink.LinkList() to retrieve all links. Inside the function we can iterate over links and check if link.MasterIndex match bridge device index. The function can return list of netlink.Link objects so outer code will not need to retrieve each link individually.

The main disadvantage of the proposed implementation is that we will need to retrieve information about all links from netlink, but I'm not sure how critical it is.

WDYT?

Reply:
i agree.  going to do remove this function and create a netlink version in  netlink.go